### PR TITLE
SAA-700: Add optional isCancelled, created, createdBy, updated, updatedBy to migrate appointments endpoint

### PR DESCRIPTION
**Pull request by iterative-it**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt [hunk 1]
```diff
@@ -59,41 +62,44 @@ class AppointmentService(
...
   fun migrateAppointment(request: AppointmentMigrateRequest, principal: Principal) =
-
     mapOf(request.prisonerNumber!! to request.bookingId.toString()).let { prisonerBookings ->
       buildValidAppointmentEntity(
+        appointmentType = AppointmentType.INDIVIDUAL,
         prisonCode = request.prisonCode!!,
+        prisonerNumbers = listOf(request.prisonerNumber),
+        prisonerBookings = prisonerBookings,
         categoryCode = request.categoryCode,
+        appointmentDescription = request.comment?.let { if (it.length <= 40) it else null },
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt:90]: Not necessary but you could do this here ...

`request.comment?.takeIf { it.length <= 40 }`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt [hunk 2]
```diff
@@ -189,6 +204,11 @@ class AppointmentService(
                 ),
               )
             }
+            if (isCancelled) {
+              cancelled = updated ?: created
+              cancellationReason = appointmentCancellationReasonRepository.findOrThrowNotFound(2)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt:209]: What is '2'?  Suggest pulling this out into a constant perhaps.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt (hunk 1 above)
Comment (iterative-it) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt]: Like it! Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt (hunk 2 above)
Comment (iterative-it) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentService.kt]: Yeah that was bad. Added constant

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Pull request merged**